### PR TITLE
Add detector jobs and review/create pages

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:*

*Description of changes:*

**This PR is 3 of 9 related to the new single flow changes. We will merge all changes into the single-flow-development branch 
first, and finally merge into main.**

**Please ignore test-related code for now, this is currently being refactored and updated, will be addressed in a later PR**

This PR adds the detector jobs and review & create pages (steps 3 & 4 in the creation flow, respectively). Specifically, the changes include:
- Adds the new `DetectorJobs` page, which allows the user to pick between starting a real-time detector and/or a historical analysis, all based off of the same configuration. Has consistent validation with the rest of the create flow pages.
- Adds the new `ReviewAndCreate` page, which allows the user to look over all of the detector configuration settings they've selected, and edit buttons to return to the corresponding step if they want to make changes. Has consistent validation with the rest of the create flow pages, and will kick off a real-time job and/or historical task if the user has selected to do so.
- `ReviewAndCreate` page components - the `AdditionalSettings`, `DetectorDefinitionFields` and `ModelConfigurationFields` are similar but refactored versions of those currently shown in the `DetectorConfig` configuration page. The new `FilterDisplayList` component allows for rendering a list of all filters created based on the new filter data model (custom labels, custom filters, simple filters). The new `DetectorScheduleFields` component shows what the user has selected regarding starting a real-time and/or historical detector upon creation.
- Updates and refactors relevant helper functions to convert Formik fields to detector

Screenshot (Detector Jobs page):
screencapture-localhost-5601-app-opendistro-anomaly-detection-kibana-2021-04-05-11_46_11

Screenshot (Review & Create page):
screencapture-localhost-5601-app-opendistro-anomaly-detection-kibana-2021-04-05-11_45_48

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/DetectorJobs/components/RealTimeJob/RealTimeJob.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+              label="Start real-time detector automatically (recommended)"
+              checked={enabled}
+              //disabled={noCategoryFields}
+              onChange={() => {
+                if (!enabled) {
+                  props.setRealTime(true);
+                }
+                if (enabled) {
+                  props.setRealTime(false);
+                  //form.setFieldValue('realTime', true);
```
Comment (yizheliu-amazon) [inline public/pages/DetectorJobs/components/RealTimeJob/RealTimeJob.tsx]: unneeded comment.

File: public/pages/DetectorJobs/containers/DetectorJobs.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          //@ts-ignore
+          props.setStep(4);
+        } else {
+          // TODO: can add focus to all components or possibly customize error message too
+          core.notifications.toasts.addDanger(
+            'One or more input fields is invalid'
+          );
+        }
+      });
+    } catch (e) {
```
Comment (yizheliu-amazon) [inline public/pages/DetectorJobs/containers/DetectorJobs.tsx]: same comment as [comment](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/389#discussion_r608003742) in another PR #389 .

File: public/pages/ReviewAndCreate/components/ModelConfigurationFields/ModelConfigurationFields.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+interface ModelConfigurationFieldsProps {
+  detector: Detector;
+  onEditModelConfiguration(): void;
+}
+
+interface ModelConfigurationFieldsState {
+  showCodeModel: boolean[];
+  sortField: string;
+  sortDirection: 'asc' | 'desc';
```
Comment (yizheliu-amazon) [inline public/pages/ReviewAndCreate/components/ModelConfigurationFields/ModelConfigurationFields.tsx]: maybe use [SORT_DIRECTION](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/ec3fdcca977a6a00cef1903140606ff23b51d9f4/server/utils/constants.ts#L42)?

File: public/pages/DetectorJobs/components/RealTimeJob/RealTimeJob.tsx (hunk 1 above)
Comment (ohltyler) [reply public/pages/DetectorJobs/components/RealTimeJob/RealTimeJob.tsx]: will remove

File: public/pages/DetectorJobs/containers/DetectorJobs.tsx (hunk 2 above)
Comment (ohltyler) [reply public/pages/DetectorJobs/containers/DetectorJobs.tsx]: removed

File: public/pages/ReviewAndCreate/components/ModelConfigurationFields/ModelConfigurationFields.tsx (hunk 3 above)
Comment (ohltyler) [reply public/pages/ReviewAndCreate/components/ModelConfigurationFields/ModelConfigurationFields.tsx]: Good catch, will change to use that.

**Review by yizheliu-amazon — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**